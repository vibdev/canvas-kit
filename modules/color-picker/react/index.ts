import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
