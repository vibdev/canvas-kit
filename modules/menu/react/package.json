{
  "name": "@workday/canvas-kit-react-menu",
  "version": "0.0.24",
  "description": "A container for navigation or action items",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "menu"
  ],
  "devDependencies": {
    "@material-ui/core": "^3.9.2",
    "@types/react-transition-group": "^2.0.14",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "react-transition-group": "^2.5.0"
  },
  "dependencies": {
    "@types/uuid": "^3.4.4",
    "@workday/canvas-kit-react-card": "^0.3.23",
    "@workday/canvas-kit-react-common": "^0.7.5",
    "@workday/canvas-kit-react-core": "^1.10.6",
    "@workday/canvas-kit-react-icon": "^1.7.7",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12",
    "uuid": "^3.3.2"
  },
  "peerDependencies": {
    "react": ">= 16.3 < 17"
  }
}
