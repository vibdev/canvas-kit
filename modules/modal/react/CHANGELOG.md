# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/modal/compare/@workday/canvas-kit-react-modal@0.0.1...@workday/canvas-kit-react-modal@0.0.2) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal





## 0.0.1 (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal
