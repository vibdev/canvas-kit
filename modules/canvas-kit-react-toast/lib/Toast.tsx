import * as React from 'react';
import Popup, {PopupPadding} from '@workday/canvas-kit-react-popup';
import {type, spacing} from '@workday/canvas-kit-react-core';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {TransformOrigin} from '@workday/canvas-kit-react-common';
import styled from 'react-emotion';

export interface ToastProps {
  handleClose?: () => void;
  handleActionLink?: () => void;
  actionLinkText?: string;
  toastIcon?: CanvasSystemIcon;
  iconColor?: string;
  transformOrigin?: TransformOrigin;
}
const toastWidth = 360;

const ToastContentContainer = styled('div')({
  display: 'flex',
});
const ToastContainer = styled('div')<Pick<ToastProps, 'handleClose'>>(
  {
    width: toastWidth,
  },
  ({handleClose}) => ({
    [`${ToastContentContainer}`]: {
      marginRight: handleClose ? spacing.m : 0,
    },
  })
);

const ToastIconContainer = styled('div')({
  marginRight: spacing.s,
});

const ActionLink = styled('a')({
  ...type.variant.link,
  display: 'flex',
  marginTop: spacing.xxxs,
});

export default class Toast extends React.Component<ToastProps> {
  public render() {
    const {
      handleClose,
      handleActionLink,
      actionLinkText,
      toastIcon,
      iconColor,
      transformOrigin,
      ...otherProps
    } = this.props;
    return (
      <ToastContainer {...otherProps} handleClose={handleClose}>
        <Popup transformOrigin={transformOrigin} padding={PopupPadding.s} handleClose={handleClose}>
          <ToastContentContainer>
            {toastIcon && (
              <ToastIconContainer>
                <SystemIcon color={iconColor} colorHover={''} icon={toastIcon} />
              </ToastIconContainer>
            )}
            <div>
              {this.props.children}
              {handleActionLink && (
                <ActionLink onClick={handleActionLink}>{actionLinkText}</ActionLink>
              )}
            </div>
          </ToastContentContainer>
        </Popup>
      </ToastContainer>
    );
  }
}
