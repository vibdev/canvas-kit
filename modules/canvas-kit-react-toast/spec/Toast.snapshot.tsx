import * as React from 'react';
import Toast from '../lib/Toast';
import * as renderer from 'react-test-renderer';
import {checkIcon} from '@workday/canvas-system-icons-web';

describe('Toast Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Toast>Your data has been updated.</Toast>);
    expect(component).toMatchSnapshot();
  });
  test('renders with a close button', () => {
    const component = renderer.create(
      <Toast handleClose={jest.fn()}>Your data has been updated.</Toast>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with an icon', () => {
    const component = renderer.create(
      <Toast toastIcon={checkIcon} handleClose={jest.fn()}>
        Success
      </Toast>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with an action link', () => {
    const component = renderer.create(
      <Toast actionLinkText={'View More'} handleActionLink={jest.fn()} handleClose={jest.fn()}>
        Process failed.
      </Toast>
    );
    expect(component).toMatchSnapshot();
  });
});
