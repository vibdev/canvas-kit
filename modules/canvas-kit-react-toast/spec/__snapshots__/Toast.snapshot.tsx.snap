// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Snapshots renders as expected 1`] = `
.emotion-5 {
  width: 360px;
}

.emotion-4 {
  position: relative;
  background-color: #ffffff;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-3 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-5 emotion-6"
>
  <div
    className="emotion-4"
    role="dialog"
  >
    <div
      className="emotion-3"
      width={360}
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-0 emotion-1"
        >
          <div
            style={
              Object {
                "marginRight": 0,
              }
            }
          >
            Your data has been updated.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with a close button 1`] = `
.emotion-9 {
  width: 360px;
}

.emotion-8 {
  position: relative;
  background-color: #ffffff;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-7 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-9 emotion-10"
>
  <div
    className="emotion-8"
    role="dialog"
  >
    <div
      className="emotion-3"
    >
      <button
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-7"
      width={360}
    >
      <div
        className="emotion-6"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <div
            style={
              Object {
                "marginRight": "24px",
              }
            }
          >
            Your data has been updated.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with an action link 1`] = `
.emotion-15 {
  width: 360px;
}

.emotion-14 {
  position: relative;
  background-color: #ffffff;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-13 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-12 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-8 {
  margin-left: -16px;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 3px;
  border: 0px;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  margin: 0 8px;
  min-width: auto;
  background-color: transparent;
  border-color: transparent;
  color: #0875e1;
  height: 32px;
  padding: 0 8px;
}

.emotion-6:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-6:disabled,
.emotion-6:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-6:hover:not([disabled]) {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6 span .wd-icon-fill,
.emotion-6 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #0875e1;
}

.emotion-6:focus {
  background-color: #d7eafc;
  color: #0875e1;
}

.emotion-6:focus span .wd-icon-fill,
.emotion-6:focus span .wd-icon-accent {
  fill: #0875e1;
}

.emotion-6:active,
.emotion-6:focus:active,
.emotion-6:hover:active {
  background-color: transparent;
  border-color: transparent;
  color: #005cb9;
}

.emotion-6:active span .wd-icon-fill,
.emotion-6:focus:active span .wd-icon-fill,
.emotion-6:hover:active span .wd-icon-fill,
.emotion-6:active span .wd-icon-accent,
.emotion-6:focus:active span .wd-icon-accent,
.emotion-6:hover:active span .wd-icon-accent {
  fill: #005cb9;
}

.emotion-6:hover {
  border-color: transparent;
  color: #005cb9;
}

.emotion-6:hover span .wd-icon-fill,
.emotion-6:hover span .wd-icon-accent {
  fill: #005cb9;
}

.emotion-6:disabled,
.emotion-6:active:disabled,
.emotion-6:focus:disabled,
.emotion-6:hover:disabled {
  background-color: transparent;
  border-color: transparent;
  color: rgba(8,117,225,0.5);
}

.emotion-6:disabled span .wd-icon-fill,
.emotion-6:active:disabled span .wd-icon-fill,
.emotion-6:focus:disabled span .wd-icon-fill,
.emotion-6:hover:disabled span .wd-icon-fill,
.emotion-6:disabled span .wd-icon-accent,
.emotion-6:active:disabled span .wd-icon-accent,
.emotion-6:focus:disabled span .wd-icon-accent,
.emotion-6:hover:disabled span .wd-icon-accent {
  fill: rgba(8,117,225,0.5);
}

.emotion-4 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0px;
  padding: 0px;
}

.emotion-4:hover:active {
  background-color: transparent;
}

<div
  className="emotion-15 emotion-16"
>
  <div
    className="emotion-14"
    role="dialog"
  >
    <div
      className="emotion-3"
    >
      <button
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-13"
      width={360}
    >
      <div
        className="emotion-12"
      >
        <div
          className="emotion-10 emotion-11"
        >
          <div
            style={
              Object {
                "marginRight": "24px",
              }
            }
          >
            Process failed.
            <div
              className="emotion-8 emotion-9"
            >
              <button
                className="emotion-6 emotion-7"
                onClick={[MockFunction]}
              >
                <span
                  className="emotion-4 emotion-5"
                >
                  View More
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with an icon 1`] = `
.emotion-11 {
  width: 360px;
}

.emotion-10 {
  position: relative;
  background-color: #ffffff;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-9 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-8 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-5 {
  display: inline-block;
  margin-right: 16px;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-5:hover .wd-icon-fill {
  fill: px;
}

.emotion-5 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-5:hover .wd-icon-accent {
  fill: px;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-11 emotion-12"
>
  <div
    className="emotion-10"
    role="dialog"
  >
    <div
      className="emotion-3"
    >
      <button
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-9"
      width={360}
    >
      <div
        className="emotion-8"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <span
            className="emotion-4 emotion-5"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
              }
            }
          />
          <div
            style={
              Object {
                "marginRight": "24px",
              }
            }
          >
            Success
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
